python/cris_import.py

```diff
@@ -0,0 +1,257 @@
+        # get schema years that match this extract ID
+        schema_years = list(
+            set([f["schema_year"] for f in files_todo if f["extract_id"] == extract_id])
+        )
+        schema_years.sort()
+        for schema_year in schema_years:
+            for table_name in ["crashes", "units", "persons"]:
+                file = next(
+                    (
+                        f
+                        for f in files_todo
+                        if f["extract_id"] == extract_id
+                        and f["schema_year"] == schema_year
+                        and table_name.startswith(f["table_name"])
+                    ),
+                    None,
+                )
+
+                if not file:
+                    raise (
+                        f"No {table_name} file found in extract. This should never happen!"
+                    )
+
+                print(f"processing {table_name}")
+
+                records = handle_empty_strings(
+                    remove_unsupported_columns(
+                        lower_case_keys(load_csv(file["path"])), table_name
+                    )
+                )
+
+                if table_name == "crashes":
+                    handle_crash_date_time(records)
+
+                # annoying redundant branch to combine primary persons and persons
+                if table_name == "persons":
+
+                    p_person_file = next(
+                        (
+                            f
+                            for f in files_todo
+                            if f["extract_id"] == extract_id
+                            and f["schema_year"] == schema_year
+                            and f["table_name"].startswith("primaryperson")
+                        ),
+                        None,
+                    )
+
+                    for p in records:
+                        p["is_primary"] = False
+
+                    pp_records = handle_empty_strings(
+                        remove_unsupported_columns(
+                            lower_case_keys(load_csv(p_person_file["path"])),
+                            table_name,
+                        )
+                    )
+                    for pp in pp_records:
+                        pp["is_primary"] = True
+                    records = pp_records + records
+
+                upsert_mutation = make_upsert_mutation(table_name)
+
+                for chunk in chunks(records, UPLOAD_BATCH_SIZE):
+                    print(f"uploading {len(chunk)} {table_name}...")
+                    start_time = time.time()
+                    res = make_hasura_request(
+                        endpoint=HASURA_ENDPOINT,
+                        query=upsert_mutation,
+                        variables={"objects": chunk},
+                    )
+                    print(f"Done in {round(time.time() - start_time, 3)} seconds")
+
+    print(f"Entire import completed in {round((time.time() - overall_start_tme)/60, 2)} minutes")
```

Comment:
these prints were such a nice touch :sparkles: my entire import running this script was completed in just over 30 seconds :rocket: